### PR TITLE
Add mapping command set

### PR DESCRIPTION
**Pull request by lamchenghou**
Link to issue: #109 

# Summary of changes 
- Add `mapping list/add/delete` commands and all relevant logic, command, parser, storage, model, ui support
- Dependency deletion block

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -225,6 +227,20 @@ public static String parseNoteTags(String tags) throws ParseException {
...
+     */
+    public static MappingMiscInformation parseMappingMiscInformation(String mappingMiscInformation)
+            throws ParseException {
+        requireNonNull(mappingMiscInformation);
+        String trimmedMappingMiscInformation = mappingMiscInformation.trim();
+        if (!MappingMiscInformation.isValidMappingMiscInformation(trimmedMappingMiscInformation)) {
+            throw new ParseException(MappingMiscInformation.MESSAGE_CONSTRAINTS);
+        }
+        return new MappingMiscInformation(trimmedMappingMiscInformation);
+    }
```
Comment (dlathyun) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:242]: Can I ask what happens if it is empty string ""?

**Review by dlathyun — COMMENTED** [review]
LGTM except for few parts that I am unsure of, such as when MappingMiscInformation is an empty string. Also I am not really familiar with UI part haha, maybe someone else can help for that part?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (lamchenghou) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:242]: I think currently empty string won't be accepted, like for all parameters.

Empty string is not allowed currently for MappingMiscInformation.

Will make changes soon to allow optional parameter for this mapping purpose.

File: src/main/java/seedu/address/model/mapping/Mapping.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * Every field must be present and not null
+     */
+    public Mapping(LocalCode lc, UniversityName un, PartnerCode pc, MappingMiscInformation mi) {
+        requireAllNonNull(lc, pc, un, mi);
+        localCode = lc;
+        partnerCode = pc;
+        universityName = un;
+        miscInformation = mi;
+    }
```
[the remark this answers is not in the document]
Comment (lamchenghou) [reply src/main/java/seedu/address/model/mapping/Mapping.java:35]: Shld be ok.

**Review by tjingsheng — APPROVED** [review]
LGTM

**Pull request merged**